packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -257,323 +231,412 @@ describe("Garbage Collection Tests", () => {
-
-            // Run garbage collection on the default GC data where everything is referenced.
-            await garbageCollector.collectGarbage({ runGC: true });
-
-            // Validate that no inactive events are logged yet.
-            await updateAllNodesAndRunGC(garbageCollector);
-            validateNoInactiveEvents();
-
-            // Expire the unreferenced timer (if any).
-            clock.tick(inactiveTimeoutMs + 1);
+        const tests = (
+            timeout: number,
+            revivedEventName: string,
+            changedEventName: string,
+            loadedEventName: string,
+            snapshotCacheExpiryMs?: number,
+            deleteEventName?: string,
+        ) => {
+            // Validates that no unexpected event has been fired.
+            function validateNoUnexpectedEvents() {
+                assert(
+                    !mockLogger.matchAnyEvent([
+                        { eventName: revivedEventName },
+                        { eventName: changedEventName },
+                        { eventName: loadedEventName },
+                        { eventName: deleteEventName },
+                    ]),
+                    "unexpected events logged",
+                );
+            }
 
-            // Validate that no inactive events are logged since everything is referenced.
-            await updateAllNodesAndRunGC(garbageCollector);
-            validateNoInactiveEvents();
-        });
+            // Returns a dummy snapshot tree to be built upon.
+            const getDummySnapshotTree = (): ISnapshotTree => {
+                return {
+                    blobs: {},
+                    trees: {},
+                };
+            };
 
-        it("generates events when inactive node is changed or revived", async () => {
-            const garbageCollector = createGarbageCollector();
+            const createGCOverride = (
+                baseSnapshot?: ISnapshotTree,
+                gcBlobsMap?: Map<string, IGarbageCollectionState | IGarbageCollectionDetailsBase>,
+            ) => {
+                return createGarbageCollector({ baseSnapshot, snapshotCacheExpiryMs }, gcBlobsMap);
+            };
 
-            // Remove node 2's reference from node 1. This should make node 2 and node 3 unreferenced.
-            defaultGCData.gcNodes[nodes[1]] = [];
+            it("doesn't generate events for referenced nodes", async () => {
+                const garbageCollector = createGCOverride();
 
-            await garbageCollector.collectGarbage({ runGC: true });
+                // Run garbage collection on the default GC data where everything is referenced.
+                await garbageCollector.collectGarbage({});
 
-            // Validate that no inactive events are logged yet.
-            await updateAllNodesAndRunGC(garbageCollector);
-            validateNoInactiveEvents();
+                // Advance the clock just before the timeout and validate no events are generated.
```

Comment:
Consider doing something similar to `closeCalledAfterExactTicks` helper used in session expiry tests?

Reply:
There are only 2 instances of this pattern so it doesn't really matter that much to inline it.